torch_xla/csrc/tensor.cpp

```diff
@@ -114,6 +114,14 @@ XLATensor::XLATensor(torch::lazy::Value ir_value,
                      c10::optional<at::ScalarType> logical_element_type)
     : XLATensor(std::make_shared<Data>(std::move(ir_value), device,
                                        logical_element_type)) {
+  // Preserve sharding if a new tensor is created from a sharded IR node.
+  if (CurrentIrValue()) {
+    auto* xla_node = dynamic_cast<XlaNode*>(CurrentIrValue().node.get());
+    if (xla_node->GetSharding()) {
+      ShardingSpec sharding = ShardingSpec{*xla_node->GetSharding()};
+      SetShardingSpec(sharding);
+    }
+  }
```

Comment:
I think I asked in a previous review but I forgot if you answered. What if passed in `ir_value` has `ShardingSpec`, is that impossible? If it is impossible can we assert on that?

Reply:
oh OK I misread, this happened after you assign the `ir_value` to self, so `CurrentIrValue` will return `ir_value`, this make sense now..